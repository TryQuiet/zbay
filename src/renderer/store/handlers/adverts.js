import BigNumber from 'bignumber.js'

import identitySelectors from '../selectors/identity'
import channelSelectors from '../selectors/channel'
import directMessageChannelSelectors from '../selectors/directMessageChannel'
import offersHandlers from '../../store/handlers/offers'
import { messages } from '../../zbay'
import { getClient } from '../../zcash'
import notificationsHandlers from './notifications'
import directMessagesQueueHandlers, {
  checkConfirmationNumber
} from './directMessagesQueue'
import { errorNotification, successNotification } from './utils'
import operationsHandlers, { operationTypes } from './operations'
import contactsHandlers from './contacts'
import { messageType } from '../../../shared/static'

const handleSend = ({ values }) => async (dispatch, getState) => {
  const data = {
    tag: values.tag,
    background: values.background.toString(),
    title: values.title,
    provideShipping: values.shippingInfo ? '1' : '0',
    amount: values.usd,
    description: values.description
  }
  const identityAddress = identitySelectors.address(getState())
  const channel =
    channelSelectors.data(getState()) &&
    channelSelectors.data(getState()).toJS()
  const directChannelAddress = directMessageChannelSelectors.targetRecipientAddress(
    getState()
  )
  const privKey = identitySelectors.signerPrivKey(getState())
  const message = messages.createMessage({
    messageData: {
      type: messageType.AD,
      data: data,
      spent: channel ? channel.advertFee.toString() : '0'
    },
    privKey
  })
  const transfer = await messages.messageToTransfer({
    message,
    address: channel ? channel.address : directChannelAddress,
    identityAddress
  })
  try {
    const opId = await getClient().payment.send(transfer)
    await dispatch(
      operationsHandlers.epics.observeOperation({
        opId,
        type: channel
          ? operationTypes.pendingMessage
          : operationTypes.pendingDirectMessage,
        meta: {
          message: message,
          channelId: channel ? channel.id : 'none',
          recipientAddress: channel ? 'none' : directChannelAddress,
          saveAdvert: !channel
        },
        checkConfirmationNumber: channel ? null : checkConfirmationNumber
      })
    )
    dispatch(
      notificationsHandlers.actions.enqueueSnackbar(
        successNotification({ message: 'Offer successfully posted' })
      )
    )
  } catch (err) {
    console.log(err)
    dispatch(
      notificationsHandlers.actions.enqueueSnackbar(
        errorNotification({
          message: "Couldn't send the message, please check node connection."
        })
      )
    )
  }
}

const handleSendTransfer = ({ values, history, payload }) => async (
  dispatch,
  getState
) => {
  const shippingData = identitySelectors.shippingData(getState())
  const privKey = identitySelectors.signerPrivKey(getState())
  const message = messages.createMessage({
    messageData: {
      type: messageType.ITEM_TRANSFER,
      data: {
        itemId: payload.id.substring(0, 64),
        tag: payload.tag,
        offerOwner: payload.offerOwner,
        shippingData: payload.provideShipping ? shippingData : null
      },
      spent: new BigNumber(values.zec)
    },
    privKey
  })
  dispatch(offersHandlers.epics.createOfferAdvert({ payload, history }))
  dispatch(
    directMessagesQueueHandlers.epics.addDirectMessage(
      {
        message,
        recipientAddress: payload.address,
        recipientUsername: payload.offerOwner
      },
      0,
      false
    )
  )
  dispatch(
    contactsHandlers.epics.updateDeletedChannelTimestamp({
      address: payload.id + payload.offerOwner,
      timestamp: 0
    })
  )
}

export const epics = {
  handleSend,
  handleSendTransfer
}

export default {
  epics
}
