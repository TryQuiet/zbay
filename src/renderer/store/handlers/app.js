import { ipcRenderer, remote } from 'electron'
import Immutable from 'immutable'
import { createAction, handleActions } from 'redux-actions'
import { actionTypes } from '../../../shared/static'
import { actionCreators } from './modals'

export const AppState = Immutable.Record(
  {
    version: null,
    transfers: Immutable.Map(),
    newUser: false,
    modalTabToOpen: null
  },
  'AppState'
)

export const initialState = AppState()

const loadVersion = createAction(actionTypes.SET_APP_VERSION, () =>
  remote.app.getVersion()
)
const setTransfers = createAction(actionTypes.SET_TRANSFERS)
const setModalTab = createAction(actionTypes.SET_CURRENT_MODAL_TAB)
const clearModalTab = createAction(actionTypes.CLEAR_CURRENT_MODAL_TAB)

export const actions = {
  loadVersion,
  setTransfers,
  setModalTab,
  clearModalTab
}

export const askForBlockchainLocation = () => async (dispatch, getState) => {
  dispatch(actionCreators.openModal('blockchainLocation')())
}

export const proceedWithSyncing = (payload) => async (dispatch, getState) => {
  ipcRenderer.send('proceed-with-syncing', payload)
  dispatch(actionCreators.closeModal('blockchainLocation')())
}

export const reducer = handleActions(
  {
    [loadVersion]: (state, { payload: version }) =>
      state.set('version', version),
    [setModalTab]: (state, { payload: tabName }) =>
      state.set('modalTabToOpen', tabName),
    [clearModalTab]: state => state.set('modalTabToOpen', null),
    [setTransfers]: (state, { payload: { id, value } }) => {
      return state.setIn(['transfers', id], value)
    }
  },
  initialState
)

export const epics = {
  askForBlockchainLocation,
  proceedWithSyncing
}

export default {
  epics,
  actions,
  reducer
}
