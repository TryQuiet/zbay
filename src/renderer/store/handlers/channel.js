import Immutable from 'immutable'
import BigNumber from 'bignumber.js'
import { createAction, handleActions } from 'redux-actions'

import history from '../../../shared/history'
import operationsHandlers, {
  operationTypes,
  PendingMessageOp
} from './operations'
import notificationsHandlers from './notifications'
import messagesQueueHandlers from './messagesQueue'
import messagesQueue from '../selectors/messagesQueue'
import messagesHandlers from './messages'
import channelsHandlers from './channels'
import offersHandlers from './offers'
import channelSelectors from '../selectors/channel'
import channelsSelectors from '../selectors/channels'
import identitySelectors from '../selectors/identity'
import logsHandlers from '../handlers/logs'
import { getClient } from '../../zcash'
import { messages } from '../../zbay'
import { errorNotification, LoaderState } from './utils'
import { channelToUri } from '../../zbay/channels'
import nodeSelectors from '../selectors/node'
import { getVault } from '../../vault'
import { messageType, actionTypes } from '../../../shared/static'

export const ChannelState = Immutable.Record(
  {
    spentFilterValue: new BigNumber(0),
    id: null,
    message: '',
    shareableUri: '',
    address: '',
    loader: LoaderState({ loading: false }),
    members: null,
    showInfoMsg: true
  },
  'ChannelState'
)

export const initialState = ChannelState()

const setSpentFilterValue = createAction(
  actionTypes.SET_SPENT_FILTER_VALUE,
  (_, value) => value
)
const setMessage = createAction(actionTypes.SET_CHANNEL_MESSAGE)
const setChannelId = createAction(actionTypes.SET_CHANNEL_ID)
const setLoading = createAction(actionTypes.SET_CHANNEL_LOADING)
const setLoadingMessage = createAction(actionTypes.SET_CHANNEL_LOADING_MESSAGE)
const setShareableUri = createAction(actionTypes.SET_CHANNEL_SHAREABLE_URI)
const setAddress = createAction(actionTypes.SET_CHANNEL_ADDRESS)
const resetChannel = createAction(actionTypes.SET_CHANNEL)

export const actions = {
  setLoading,
  setLoadingMessage,
  setSpentFilterValue,
  setMessage,
  setShareableUri,
  setChannelId,
  resetChannel
}

const loadChannel = id => async (dispatch, getState) => {
  try {
    dispatch(setChannelId(id))

    // Calculate URI on load, that way it won't be outdated, even if someone decides
    // to update channel in vault manually
    const channel = channelSelectors.data(getState()).toJS()
    const uri = await channelToUri(channel)
    dispatch(setShareableUri(uri))
    dispatch(setAddress(channel.address))
    await dispatch(clearNewMessages())
    await dispatch(updateLastSeen())
  } catch (err) {}
}
const loadOffer = (id, address) => async (dispatch, getState) => {
  try {
    await dispatch(offersHandlers.epics.updateLastSeen({ itemId: id }))
    dispatch(setChannelId(id))
    dispatch(setShareableUri(''))
    dispatch(setAddress(address))
  } catch (err) {}
}
const linkChannelRedirect = targetChannel => async (dispatch, getState) => {
  let channels = channelsSelectors.channels(getState())
  let channel = channels.data.find(
    channel => channel.get('address') === targetChannel.address
  )
  const identityId = identitySelectors.id(getState())
  const lastblock = nodeSelectors.latestBlock(getState())
  const fetchTreshold = lastblock - 2000
  if (channel) {
    history.push(`/main/channel/${channel.get('id')}`)
    return
  }
  try {
    try {
      await getVault().channels.importChannel(identityId, targetChannel)
      getClient().keys.importIVK({
        ivk: targetChannel.keys.ivk,
        rescan: 'yes',
        startHeight: fetchTreshold
      })
    } catch (error) {}
    dispatch(logsHandlers.epics.saveLogs({ type: 'APPLICATION_LOGS', payload: `Importing channel ${targetChannel}` }))
    await dispatch(channelsHandlers.actions.loadChannels(identityId))
    channels = channelsSelectors.channels(getState())
    channel = channels.data.find(
      channel => channel.get('address') === targetChannel.address
    )
    await dispatch(messagesHandlers.epics.fetchMessages(channel))
    dispatch(
      notificationsHandlers.actions.enqueueSnackbar({
        message: `Successfully imported channel ${targetChannel.name}`,
        options: {
          variant: 'success'
        }
      })
    )
    history.push(`/main/channel/${channel.get('id')}`)
  } catch (err) {
    console.log(err)
  }
}
const sendOnEnter = (event, resetTab) => async (dispatch, getState) => {
  if (resetTab) {
    resetTab(0)
  }
  const enterPressed = event.nativeEvent.keyCode === 13
  const shiftPressed = event.nativeEvent.shiftKey === true
  const channel = channelSelectors.data(getState()).toJS()
  const messageToSend = channelSelectors.message(getState())
  const currentMessage = messagesQueue
    .queue(getState())
    .find(dm => dm.get('channelId') === channel.id)
  if (enterPressed && !shiftPressed) {
    event.preventDefault()
    const privKey = identitySelectors.signerPrivKey(getState())
    let message
    if (currentMessage !== undefined) {
      message = messages.createMessage({
        messageData: {
          type: messageType.BASIC,
          data:
            currentMessage.get('message').get('message') + '\n' + messageToSend
        },
        privKey: privKey
      })
    } else {
      message = messages.createMessage({
        messageData: {
          type: messageType.BASIC,
          data: messageToSend
        },
        privKey: privKey
      })
    }
    dispatch(
      messagesQueueHandlers.epics.addMessage({ message, channelId: channel.id })
    )
    dispatch(setMessage(''))
  }
}
const sendChannelSettingsMessage = ({
  address,
  minFee = '0',
  onlyRegistered = '0'
}) => async (dispatch, getState) => {
  const identityAddress = identitySelectors.address(getState())
  const owner = identitySelectors.signerPubKey(getState())
  const privKey = identitySelectors.signerPrivKey(getState())
  const message = messages.createMessage({
    messageData: {
      type: messageType.CHANNEL_SETTINGS,
      data: {
        owner,
        minFee,
        onlyRegistered
      }
    },
    privKey: privKey
  })
  const transfer = await messages.messageToTransfer({
    message,
    address: address,
    identityAddress
  })
  try {
    await getClient().payment.send(transfer)
  } catch (err) {
    notificationsHandlers.actions.enqueueSnackbar(
      dispatch(
        errorNotification({
          message: "Couldn't create channel, please check node connection."
        })
      )
    )
  }
}

const resendMessage = messageData => async (dispatch, getState) => {
  dispatch(operationsHandlers.actions.removeOperation(messageData.id))
  const identityAddress = identitySelectors.address(getState())
  const channel = channelSelectors.data(getState()).toJS()
  const privKey = identitySelectors.signerPrivKey(getState())
  const message = messages.createMessage({
    messageData: {
      type: messageData.type,
      data: messageData.message,
      spent: '0'
    },
    privKey
  })
  const transfer = await messages.messageToTransfer({
    message,
    address: channel.address,
    identityAddress
  })
  try {
    const opId = await getClient().payment.send(transfer)
    await dispatch(
      operationsHandlers.epics.observeOperation({
        opId,
        type: operationTypes.pendingMessage,
        meta: PendingMessageOp({
          channelId: channel.id,
          message: Immutable.fromJS(message)
        })
      })
    )
  } catch (err) {
    notificationsHandlers.actions.enqueueSnackbar(
      errorNotification({
        message: "Couldn't send the message, please check node connection."
      })
    )
  }
}

const updateLastSeen = () => async (dispatch, getState) => {
  const channelId = channelSelectors.channelId(getState())
  return dispatch(channelsHandlers.epics.updateLastSeen({ channelId }))
}

const clearNewMessages = () => async (dispatch, getState) => {
  const channelId = channelSelectors.channelId(getState())
  dispatch(messagesHandlers.actions.cleanNewMessages({ channelId }))
}

export const epics = {
  sendOnEnter,
  loadChannel,
  resendMessage,
  clearNewMessages,
  updateLastSeen,
  loadOffer,
  sendChannelSettingsMessage,
  linkChannelRedirect
}

// TODO: we should have a global loader map
export const reducer = handleActions(
  {
    [setLoading]: (state, { payload: loading }) =>
      state.setIn(['loader', 'loading'], loading),
    [setLoadingMessage]: (state, { payload: message }) =>
      state.setIn(['loader', 'message'], message),
    [setSpentFilterValue]: (state, { payload: value }) =>
      state.set('spentFilterValue', new BigNumber(value)),
    [setMessage]: (state, { payload: value }) => state.set('message', value),
    [setChannelId]: (state, { payload: id }) => state.set('id', id),
    [setShareableUri]: (state, { payload: uri }) =>
      state.set('shareableUri', uri),
    [setAddress]: (state, { payload: address }) =>
      state.set('address', address),
    [resetChannel]: () => initialState
  },
  initialState
)

export default {
  reducer,
  epics,
  actions
}
