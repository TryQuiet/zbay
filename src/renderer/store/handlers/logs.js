import Immutable from 'immutable'
import { createAction, handleActions } from 'redux-actions'
import { ipcRenderer } from 'electron'
import { actionTypes } from '../../../shared/static'

export const Logs = Immutable.Record(
  {
    transactionLogs: [],
    applicationLogs: [],
    nodeLogs: [],
    islogsFileLoaded: false,
    isLogWindowOpened: false
  },
  'logs'
)

export const initialState = Logs()

const setTransactionLogs = createAction(actionTypes.SET_TRANSACTIONS_LOGS)
const setApplicationLogs = createAction(actionTypes.SET_APPLICATIONS_LOGS)
const setNodeLogs = createAction(actionTypes.SET_NODE_LOGS)
const setLogWindowOpened = createAction(actionTypes.SET_LOG_WINDOW_OPENED)

const loadTargetLogs = () => async (dispatch, getState) => {
  ipcRenderer.send('load-logs')
  dispatch(setLogWindowOpened(true))
}

const closeLogsWindow = () => async (dispatch, getState) => {
  ipcRenderer.send('disable-load-logs')
  dispatch(setLogWindowOpened(false))
}

const saveLogs = (data) => async (dispatch, getState) => {
  ipcRenderer.send('save-to-log-file', data)
}

export const actions = {
  setTransactionLogs,
  setApplicationLogs,
  setNodeLogs,
  setLogWindowOpened
}

export const epics = {
  loadTargetLogs,
  closeLogsWindow,
  saveLogs
}

export const reducer = handleActions(
  {
    [setTransactionLogs]: (state, { payload: transactionLogs }) => state.setIn(['transactionLogs'], transactionLogs),
    [setApplicationLogs]: (state, { payload: ApplicationLogs }) => state.setIn(['applicationLogs'], ApplicationLogs),
    [setNodeLogs]: (state, { payload: setNodeLogs }) => state.setIn(['nodeLogs'], setNodeLogs),
    [setLogWindowOpened]: (state, { payload: logPanelStatus }) => state.setIn(['isLogWindowOpened'], logPanelStatus)
  },
  initialState
)

export default {
  actions,
  epics,
  reducer
}
