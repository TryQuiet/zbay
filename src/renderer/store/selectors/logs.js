import { createSelector } from 'reselect'

const store = s => s

export const logs = createSelector(store, state => state.get('logsData'))

const applicationLogs = createSelector(logs, a => a.applicationLogs)
const transactionsLogs = createSelector(logs, a => a.transactionLogs)
const nodeLogs = createSelector(logs, a => a.nodeLogs)
const islogsFileLoaded = createSelector(logs, a => a.islogsFileLoaded)
const isLogWindowOpened = createSelector(logs, a => a.isLogWindowOpened)

export default {
  applicationLogs,
  transactionsLogs,
  nodeLogs,
  islogsFileLoaded,
  isLogWindowOpened
}
