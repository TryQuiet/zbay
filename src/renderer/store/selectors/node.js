import { DateTime, Interval } from 'luxon'
import { createSelector } from 'reselect'
import BigNumber from 'bignumber.js'

const store = s => s

export const node = createSelector(store, state => state.get('node'))

const currentBlock = createSelector(node, n => n.currentBlock)
const latestBlock = createSelector(node, n => n.latestBlock)
const connections = createSelector(node, n => n.connections)
const status = createSelector(node, n => n.status)
const uptime = createSelector(
  node,
  n => Interval
    .fromDateTimes(DateTime.fromISO(n.startedAt), DateTime.utc())
    .toDuration(['days', 'hours', 'minutes', 'seconds'])
    .normalize()
    .toObject()
)
const percentSynced = createSelector(
  [currentBlock, latestBlock],
  (current, latest) => {
    if (!latest.isZero()) {
      return current.dividedBy(latest).multipliedBy(100).toFixed(0, BigNumber.ROUND_DOWN)
    }
    return null
  }
)
const network = createSelector(node, n => {
  if (parseInt(process.env.ZBAY_IS_TESTNET) === 1) {
    return 'testnet'
  } else {
    return 'mainnet'
  }
})

const isConnected = createSelector(status, s => ['healthy', 'syncing'].includes(s))

const loader = createSelector(node, n => n.bootstrapLoader)
const fetching = createSelector(node, n => n.fetchingStatus)
const bootstrapping = createSelector(loader, n => n.loading)
const bootstrappingMessage = createSelector(loader, n => n.message)
const fetchingPart = createSelector(fetching, n => n.part)
const fetchingSize = createSelector(fetching, n => n.sizeLeft)
const fetchingStatus = createSelector(fetching, n => n.fetchingStatus)
const fetchingSpeed = createSelector(fetching, n => n.fetchingSpeed)
const fetchingEndTime = createSelector(fetching, n => n.fetchingEndTime)
const rescanningProgress = createSelector(fetching, n => n.rescanningProgress)
const isFetching = createSelector(fetching, n => n.isFetching)
const isRescanningMonitorStarted = createSelector(fetching, n => n.isRescanningMonitorStarted)
const isRescanningInitialized = createSelector(fetching, n => n.isRescanningInitialized)
const guideStatus = createSelector(fetching, n => n.guideStatus)

export default {
  node,
  currentBlock,
  latestBlock,
  status,
  uptime,
  connections,
  network,
  percentSynced,
  isConnected,
  bootstrapping,
  bootstrappingMessage,
  fetchingPart,
  fetchingSize,
  fetchingStatus,
  fetchingSpeed,
  isFetching,
  fetchingEndTime,
  rescanningProgress,
  isRescanningMonitorStarted,
  isRescanningInitialized,
  guideStatus
}
