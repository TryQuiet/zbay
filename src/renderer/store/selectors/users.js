import { createSelector } from 'reselect'
import identitySelectors from './identity'
const store = s => s

const users = createSelector(store, state => {
  return state.get('users')
})

const isRegisteredUsername = nickname => createSelector(users, (users) => {
  return users.toList().map(user => user.get('nickname')).includes(nickname)
})

const registeredUser = signerPubKey =>
  createSelector(users, users => users.get(signerPubKey))

const myUser = createSelector(
  users,
  identitySelectors.signerPubKey,
  (users, signerPubKey) => {
    return (
      users.get(signerPubKey) || {
        firstName: '',
        lastName: '',
        nickname: 'Anon' + signerPubKey.substring(0, 15),
        address: '',
        createdAt: 0
      }
    )
  }
)

export default {
  users,
  myUser,
  registeredUser,
  isRegisteredUsername
}
