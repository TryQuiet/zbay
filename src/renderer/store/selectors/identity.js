import BigNumber from 'bignumber.js'
import { createSelector } from 'reselect'
import * as R from 'ramda'

import { rate } from './rates'
import operationsSelectors from './operations'
import { operationTypes } from '../handlers/operations'

const store = s => s

const identity = createSelector(store, state => state.get('identity'))

const data = createSelector(identity, i => i.data)

const balance = currency =>
  createSelector(data, rate(currency), (d, rate) =>
    rate.times(new BigNumber(d.balance || 0))
  )

const lockedBalance = currency =>
  createSelector(data, rate(currency), (d, rate) =>
    rate.times(new BigNumber(d.lockedBalance || 0))
  )

const id = createSelector(data, d => d.id)
const name = createSelector(data, d => d.name)

const donationAllow = createSelector(data, d => d.donationAllow)
const freeUtxos = createSelector(data, d => d.freeUtxos)
const shieldingTax = createSelector(data, d => d.shieldingTax)
const donationAddress = createSelector(data, d => d.donationAddress)

const donation = createSelector(data, d => ({
  allow: d.donationAllow,
  address: d.donationAddress
}))
const signerPrivKey = createSelector(data, d => d.signerPrivKey)
const signerPubKey = createSelector(data, d => d.signerPubKey)

const address = createSelector(data, d => d.address)
const transparentAddress = createSelector(data, d => d.transparentAddress)
const topAddress = createSelector(
  data,
  d => d.addresses.get(0) || d.transparentAddress
)
const addresses = createSelector(data, d => d.addresses)
const topShieldedAddress = createSelector(
  data,
  d => d.shieldedAddresses.get(0) || d.address
)
const shieldedAddresses = createSelector(data, d => d.shieldedAddresses)

const _isShieldOperation = R.curry((identity, op) => {
  return (
    op.type === operationTypes.shieldBalance &&
    op.meta.from === identity.transparentAddress &&
    op.meta.to === identity.address
  )
})

const transparentBalance = createSelector(
  data,
  operationsSelectors.operations,
  (d, ops) =>
    R.compose(
      R.reduce(
        (acc, [key, val]) => acc.plus(val.meta.amount),
        new BigNumber(0)
      ),
      o => o.filter(_isShieldOperation(d))
    )(ops)
)

const loader = createSelector(identity, i => i.loader)

const shippingData = createSelector(data, d => d.shippingData)

export default {
  id,
  name,
  data,
  identity,
  address,
  transparentAddress,
  transparentBalance,
  lockedBalance,
  balance,
  loader,
  shippingData,
  signerPrivKey,
  signerPubKey,
  donationAllow,
  donationAddress,
  donation,
  shieldingTax,
  freeUtxos,
  topAddress,
  topShieldedAddress,
  addresses,
  shieldedAddresses
}
