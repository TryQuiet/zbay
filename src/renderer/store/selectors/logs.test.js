/* eslint import/first: 0 */
import Immutable from 'immutable'

import selectors from './logs'
import { Logs } from '../handlers/logs'

import create from '../create'

describe('logs selector', () => {
  let store = null
  beforeEach(() => {
    jest.clearAllMocks()
    store = create({
      initialState: Immutable.Map({
        logsData: Logs()
      })
    })
  })

  it('applicationLogs selector', async () => {
    expect(selectors.applicationLogs(store.getState())).toMatchSnapshot()
  })

  it('transactionsLogs selector', async () => {
    expect(selectors.transactionsLogs(store.getState())).toMatchSnapshot()
  })

  it('nodeLogs selector', async () => {
    expect(selectors.nodeLogs(store.getState())).toMatchSnapshot()
  })

  it('islogsFileLoaded selector', async () => {
    expect(selectors.islogsFileLoaded(store.getState())).toMatchSnapshot()
  })

  it('isLogWindowOpened selector', async () => {
    expect(selectors.isLogWindowOpened(store.getState())).toMatchSnapshot()
  })
})
