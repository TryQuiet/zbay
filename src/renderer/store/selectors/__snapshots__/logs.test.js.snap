// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logs selector applicationLogs selector 1`] = `Array []`;

exports[`logs selector isLogWindowOpened selector 1`] = `false`;

exports[`logs selector islogsFileLoaded selector 1`] = `false`;

exports[`logs selector nodeLogs selector 1`] = `Array []`;

exports[`logs selector transactionsLogs selector 1`] = `Array []`;
